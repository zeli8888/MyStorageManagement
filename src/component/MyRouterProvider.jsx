import { createBrowserRouter, RouterProvider } from 'react-router-dom';
import IngredientComponent from './IngredientComponent';
import DishComponent from './DishComponent';
import DishRecordComponent from './DishRecordComponent';
import LoginComponent from './LoginComponent';
import SessionFilter from './SessionFilter';
import HomeComponent from './HomeComponent';
import RegistrationComponent from './RegistrationComponent';
import ResetPasswordComponent from './ResetPasswordComponent';
import App from '../App';
const MyRouterProvider = () => {
    const routes = [
        {
            Component: App, // root layout route
            children: [
                {
                    path: '/',
                    Component: SessionFilter,
                    children: [
                        {
                            path: 'home',
                            Component: HomeComponent
                        },
                        {
                            path: 'foodstorage',
                            children: [
                                {
                                    path: 'ingredients',
                                    Component: IngredientComponent,
                                },
                                {
                                    path: 'dishes',
                                    Component: DishComponent,
                                },
                                {
                                    path: 'dishrecords',
                                    Component: DishRecordComponent,
                                },
                            ]
                        },
                    ]
                },
                {
                    path: '/login',
                    Component: LoginComponent,
                },
                {
                    path: '/register',
                    Component: RegistrationComponent,
                },
                {
                    path: '/reset-password',
                    Component: ResetPasswordComponent,
                },
            ],
        },
    ]
    const routerConfig = {
        basename: import.meta.env.VITE_REACT_APP_CONTEXT
    }
    const router = createBrowserRouter(routes, routerConfig);
    return (
        <RouterProvider router={router} />
    )
}

export default MyRouterProvider