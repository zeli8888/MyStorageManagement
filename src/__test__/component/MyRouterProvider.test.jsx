// MyRouterProvider.test.tsx
import { describe, it, expect, vi, beforeEach } from 'vitest';
import { render, screen } from '@testing-library/react';
import { createBrowserRouter, RouterProvider, MemoryRouter } from 'react-router-dom';
import MyRouterProvider from '../../component/MyRouterProvider';
import '@testing-library/jest-dom/vitest'

// Mock components to avoid rendering actual implementations
vi.mock('../../component/LoginComponent', () => ({ default: () => <div>Login</div> }));
vi.mock('../../component/RegistrationComponent', () => ({ default: () => <div>Register</div> }));
// Mock react-router functions
vi.mock('react-router-dom', async (importOriginal) => {
    const actual = await importOriginal();
    return {
        ...actual,
        createBrowserRouter: vi.fn(),
        RouterProvider: vi.fn(() => <div>RouterProvider</div>),
    };
});

describe('MyRouterProvider Component', () => {
    beforeEach(() => {
        vi.clearAllMocks();
        vi.stubEnv('VITE_REACT_APP_CONTEXT', '/test');
    });

    it('should create router with correct configuration', () => {
        const mockRouter = {};
        createBrowserRouter.mockReturnValue(mockRouter);

        const { asFragment } = render(<MyRouterProvider />);
        expect(asFragment()).toMatchSnapshot();

        expect(createBrowserRouter).toHaveBeenCalledWith(
            expect.anything(),
            expect.objectContaining({ basename: '/test' })
        );
        expect(RouterProvider.mock.calls[0][0]).toEqual(
            expect.objectContaining({ router: mockRouter })
        );
    });
});