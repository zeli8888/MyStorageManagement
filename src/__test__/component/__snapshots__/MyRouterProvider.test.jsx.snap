// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyRouterProvider Component > should create router with correct configuration 1`] = `
<DocumentFragment>
  <div>
    RouterProvider
  </div>
</DocumentFragment>
`;
